src/components/pill/mastercardPill.js

```diff
@@ -28,9 +29,10 @@ const MasterCardPill = forwardRef(({ isClickable, flavour, onClick, side, text,
     <Pill
       background={background}
       borderColor={background}
+      data-testid={dataTestId || "mastercard-pill"}
```

Comment:
Those `'data-testid': dataTestId` are not needed and better to not make strict such props. Just pass the rest down the component.

Reply:
```
data-testid="static-text"
{...rest}
```
if rest includes a testid it will override the previous/static one.